oss_src/lambda/pylambda_worker.cpp

```diff
@@ -81,49 +83,54 @@ int EXPORT pylambda_worker_main(const std::string& root_path,
+    __TRACK; cppipc::comm_server server(std::vector<std::string>(), "", server_address);
 
-    server.register_type<graphlab::lambda::lambda_evaluator_interface>([&](){
+    __TRACK; server.register_type<graphlab::lambda::lambda_evaluator_interface>([&](){
         if (has_shm) {
-          auto n = new graphlab::lambda::pylambda_evaluator(&shm_comm_server);
-          LOG_DEBUG_WITH_PID("creation of pylambda_evaluator with SHM complete.");
-          return n;
+          __TRACK; auto n = new graphlab::lambda::pylambda_evaluator(&shm_comm_server);
+          __TRACK; LOG_DEBUG_WITH_PID("creation of pylambda_evaluator with SHM complete.");
+          __TRACK; return n;
         } else {
-          auto n = new graphlab::lambda::pylambda_evaluator();
-          LOG_DEBUG_WITH_PID("creation of pylambda_evaluator without SHM complete.");
-          return n;
+          __TRACK; auto n = new graphlab::lambda::pylambda_evaluator();
+          __TRACK; LOG_DEBUG_WITH_PID("creation of pylambda_evaluator without SHM complete.");
+          __TRACK; return n;
         }
       });
-    server.register_type<graphlab::lambda::graph_lambda_evaluator_interface>([&](){
-        auto n = new graphlab::lambda::graph_pylambda_evaluator();
-        LOG_DEBUG_WITH_PID("creation of graph_pylambda_evaluator complete.");
-        return n;
+    
+    __TRACK; server.register_type<graphlab::lambda::graph_lambda_evaluator_interface>([&](){
+        __TRACK; auto n = new graphlab::lambda::graph_pylambda_evaluator();
+        __TRACK; LOG_DEBUG_WITH_PID("creation of graph_pylambda_evaluator complete.");
+        __TRACK; return n;
       });
 
-    LOG_DEBUG_WITH_PID("Starting server.");
-    server.start();
+    __TRACK; LOG_DEBUG_WITH_PID("Starting server.");
+    __TRACK; server.start();
 
-    wait_for_parent_exit(parent_pid);
+    __TRACK; wait_for_parent_exit(parent_pid);
 
     return 0;
 
-    /** Any exceptions happening?
+    /** Any exceptions happening?  If so, propegate back what's going
+     *  on through the error codes.
      */
   } catch (const std::string& error) {
-    logstream(LOG_ERROR) << "Internal PyLambda Error: " << error << std::endl;
-    return 103;
+    logstream(LOG_ERROR) << "Internal PyLambda Error: " << error
+                         << "; last successful line =" << _last_line  << std::endl;
+    return 1000 + _last_line;
   } catch (const std::exception& error) {
-    logstream(LOG_ERROR) << "PyLambda C++ Error: " << error.what() << std::endl;
-    return 104;
+    logstream(LOG_ERROR) << "PyLambda C++ Error: " << error.what()
+                         << "; last successful line =" << _last_line << std::endl;
+    return 2000 + _last_line;
   } catch (...) {
-    logstream(LOG_ERROR) << "Unknown PyLambda Error." << std::endl;
-    return 105;
+    logstream(LOG_ERROR) << "Unknown PyLambda Error"
+                         << "; last successful line =" << _last_line << std::endl;
+    return 3000 + _last_line;
```

Comment:
Do any of these error codes here end up being returned as exit codes? Because exit codes are only in the range 0-255